### PR TITLE
HapticFeedback: can be disabled in settings

### PR DESCRIPTION
**Pull request by danielepintore**
Hi, I've added an toggle in the settings to disable haptic feedback in the app. Now the haptic feedback is managed via the `Haptic` enum, this allows to check for the user preference so we don't need to check every time.

The usage of the Haptic enum is pretty simple, for example:
`Haptics.FEEDBACK_TYPE.vibrate()`
Where `FEEDBACK_TYPE` is the type of feedback we want

I couldn't test this in a real device so please if you can give it a check

### CODE REVIEW

**Comment by BLeeEZ** [comment]
Hi,
first of all thank you for your contribution.
Can you explain to me the use case. Is this an accessibility feature?
One remark regarding your code. I usually try avoid the use of singletons. I would prefer that the user preference must be provided as input to the vibrate method.

**Comment by danielepintore** [comment]
No i didn't think that as an accessibility feature, while i was using the app i thought if i could disable the haptic feedback because i didn't like. Just a mere preference. 
For example when you are "zapping" some songs in the library it's not handy to me have the phone vibrate each time, it become kinda awkward.
Instead for the singletons, if you like the idea of adding this feature i can fix that, just let me know.

File: Amperfy/Haptics.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        case heavy
+        @available(iOS 13.0, *)
+        case soft
+        @available(iOS 13.0, *)
+        case rigid
+        case selection
+        case oldSchool
+
+        public func vibrate() {
+            if AmperKit.shared.storage.settings.isHapticsEnabled {
```
Comment (BLeeEZ) [inline Amperfy/Haptics.swift:29]: This could be a `guard` check. This would remove one indentation.

File: Amperfy/Haptics.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        case medium
+        case heavy
+        @available(iOS 13.0, *)
+        case soft
+        @available(iOS 13.0, *)
+        case rigid
+        case selection
+        case oldSchool
+
+        public func vibrate() {
```
Comment (BLeeEZ) [inline Amperfy/Haptics.swift:28]: provide the isHapticsEnabled as method input.

**Comment by BLeeEZ** [comment]
OK, this makes sense. If you adjust the two code comments, it is ready for merge.

**Comment by danielepintore** [comment]
Ok, I've made the changes. I've also refactored the Haptics file because i noticed that was double indented, now everything should be fine.

**Comment by BLeeEZ** [comment]
I renamed "Display" to "Display & Interaction" and moved the haptic feedback into this new section. I added a description too.

**Pull request merged**